x-pack/platform/plugins/shared/actions/server/lib/categorize_http_errors.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AxiosError } from 'axios';
+import { createTaskRunError, TaskErrorSource } from '@kbn/task-manager-plugin/server';
+import type { ErrorCategorizationOverrides } from '../types';
+
+export const httpResponseUserErrorCodes = [
+  401, 402, 403, 407, 409, 412, 413, 417, 422, 423, 429, 451,
+];
+
+/**
+ * Categorizes errored actions HTTP requests against external systems, creating user errors based
+ * on the status code of the response and any overrides provided.
+ */
+export const handleActionHttpUserErrors = <T = unknown, D = unknown>(
+  error: AxiosError<T, D>,
+  overrides?: ErrorCategorizationOverrides
```

Comment:
I found the signature of the `override` a bit confusing. Can't we just always throw a user error for `httpResponseUserErrorCodes`, and the webhook can throw the extra codes? An alternative would be to change the `overrides` to something like `extraHttpCodes: string[]` that will extend the `httpResponseUserErrorCodes`.